src/protocol/prf_sharding/mod.rs

```diff
@@ -0,0 +1,704 @@
+            &(share_of_one - &inputs_required_from_previous_row.is_saturated),
+            ctx.narrow(&Step::IsSaturatedAndPrevRowNotSaturated(row_for_user)),
+            record_id,
+        )
+        .await?;
+
+    let difference_to_cap = BitDecomposed::new(
+        compute_truncated_difference_to_cap(
+            ctx.narrow(&Step::ComputeDifferenceToCap(row_for_user)),
+            record_id,
+            &saturating_sum,
+            num_trigger_value_bits,
+            num_saturating_sum_bits,
+        )
+        .await?,
+    );
+
+    let narrowed_ctx1 = ctx.narrow(&Step::ComputedCappedAttributedTriggerValueNotSaturatedCase(
+        row_for_user,
+    ));
+    let narrowed_ctx2 =
+        ctx.narrow(&Step::ComputedCappedAttributedTriggerValueJustSaturatedCase(row_for_user));
+    let capped_attributed_trigger_value = BitDecomposed::new(
+        try_join_all(
+            attributed_trigger_value
+                .iter()
+                .zip(inputs_required_from_previous_row.difference_to_cap.iter())
+                .zip(repeat(is_saturated.clone()))
+                .zip(repeat(is_saturated_and_prev_row_not_saturated))
+                .enumerate()
+                .map(
+                    |(
+                        i,
+                        (
+                            ((attributed_tv_bit, prev_row_diff_to_cap_bit), is_saturated),
+                            is_saturated_and_prev_row_not_saturated,
+                        ),
+                    )| {
+                        let c1 = narrowed_ctx1.narrow(&BitOpStep::from(i));
+                        let c2 = narrowed_ctx2.narrow(&BitOpStep::from(i));
+                        async move {
+                            let not_saturated_case = (Replicated::share_known_value(&c1, Gf2::ONE)
+                                - &is_saturated)
+                                .multiply(attributed_tv_bit, c1, record_id)
+                                .await?;
+                            let just_saturated_case = is_saturated_and_prev_row_not_saturated
+                                .multiply(prev_row_diff_to_cap_bit, c2, record_id)
+                                .await?;
+                            Ok::<_, Error>(not_saturated_case + &just_saturated_case)
+                        }
+                    },
+                ),
+        )
+        .await?,
+    );
+
+    let inputs_required_for_next_row = InputsRequiredFromPrevRow {
+        ever_encountered_a_source_event,
+        attributed_breakdown_key_bits: BitDecomposed::new(attributed_breakdown_key_bits.clone()),
+        saturating_sum,
+        is_saturated,
+        difference_to_cap,
+    };
+    let outputs_for_aggregation = CappedAttributionOutputs {
+        did_trigger_get_attributed,
+        attributed_breakdown_key_bits,
+        capped_attributed_trigger_value,
+    };
+    Ok((inputs_required_for_next_row, outputs_for_aggregation))
+}
+
+#[cfg(all(test, any(unit_test, feature = "shuttle")))]
```

Comment:
why do you need to run these tests under the randomized scheduler? there seem to be no obvious point where we call shuttle executor here

Reply:
I did that because...
...copy paste without understanding what I'm doing? 

:)